apiserver/admin_test.go

```diff
@@ -1,1252 +0,0 @@
-// Copyright 2012-2014 Canonical Ltd.
-// Licensed under the AGPLv3, see LICENCE file for details.
-
-package apiserver_test
-
-import (
-	"bytes"
-	"context"
-	"fmt"
-	"io"
-	"net/http"
-	"net/url"
-	"strconv"
-	"strings"
-	"time"
-
-	"github.com/juju/clock/testclock"
-	"github.com/juju/errors"
-	"github.com/juju/loggo/v2"
-	"github.com/juju/names/v6"
-	jc "github.com/juju/testing/checkers"
-	gc "gopkg.in/check.v1"
-
-	"github.com/juju/juju/api"
-	apimachiner "github.com/juju/juju/api/agent/machiner"
-	apiclient "github.com/juju/juju/api/client/client"
-	machineclient "github.com/juju/juju/api/client/machinemanager"
-	"github.com/juju/juju/api/client/modelconfig"
-	corecontroller "github.com/juju/juju/controller"
-	"github.com/juju/juju/core/constraints"
-	"github.com/juju/juju/core/migration"
-	coremodel "github.com/juju/juju/core/model"
-	"github.com/juju/juju/core/network"
-	"github.com/juju/juju/core/permission"
-	"github.com/juju/juju/core/user"
-	usertesting "github.com/juju/juju/core/user/testing"
-	jujuversion "github.com/juju/juju/core/version"
-	"github.com/juju/juju/domain/access"
-	accessservice "github.com/juju/juju/domain/access/service"
-	"github.com/juju/juju/internal/auth"
-	loggertesting "github.com/juju/juju/internal/logger/testing"
-	"github.com/juju/juju/internal/password"
-	coretesting "github.com/juju/juju/internal/testing"
-	"github.com/juju/juju/internal/testing/factory"
-	"github.com/juju/juju/internal/uuid"
-	jujutesting "github.com/juju/juju/juju/testing"
-	"github.com/juju/juju/rpc"
-	"github.com/juju/juju/rpc/params"
-	"github.com/juju/juju/state"
-)
-
-const (
-	clientFacadeVersion           = 8
-	machineManagerFacadeVersion   = 11
-	sshClientFacadeVersion        = 4
-	pingerFacadeVersion           = 1
-	modelManagerFacadeVersion     = 10
-	highAvailabilityFacadeVersion = 2
-)
-
-type baseLoginSuite struct {
-	jujutesting.ApiServerSuite
-	mgmtSpace *network.SpaceInfo
-}
-
-func (s *baseLoginSuite) SetUpTest(c *gc.C) {
-	s.ApiServerSuite.SetUpTest(c)
-	loggo.GetLogger("juju.apiserver").SetLogLevel(loggo.TRACE)
-
-	networkService := s.ControllerDomainServices(c).Network()
-	mgmtSpaceID, err := networkService.AddSpace(context.Background(), network.SpaceInfo{
-		Name: "mgmt01",
-	})
-	c.Assert(err, jc.ErrorIsNil)
-	s.mgmtSpace, err = networkService.Space(context.Background(), mgmtSpaceID.String())
-	c.Assert(err, jc.ErrorIsNil)
-
-	cfg := map[string]any{
-		corecontroller.JujuManagementSpace: "mgmt01",
-	}
-
-	configService := s.ControllerDomainServices(c).ControllerConfig()
-	err = configService.UpdateControllerConfig(context.Background(), cfg, nil)
-	c.Assert(err, jc.ErrorIsNil)
-}
-
-type loginSuite struct {
```

Comment:
Very reluctant to lose the user login tests here. Those don't actually use any legacy state, just the domain services that were added to the `ApiServerSuite`.

The agent tests and user tests should be in separate suites.

Reply:
These should have been re-written with the auth epic.